src/components/Layout/Header/Navigation.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import styled from 'styled-components'
+import { NavLink } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronCircleUp, faChevronCircleDown } from '@fortawesome/free-solid-svg-icons'
+
+import LinkWithPastLocation from 'components/LinkWithPastLocation'
+
+export type NavLinkKeys = 'Trade' | 'Wallet' | 'Orders'
+
+export const NavLinksWrapper = styled.div<{ $open?: boolean; $responsive: boolean }>`
+  display: flex;
+  justify-content: space-evenly;
+
+  margin: 0.5rem;
+  padding: 0.5rem;
+
+  white-space: nowrap;
+
+  a {
+    color: var(--color-text-secondary);
+    font-size: 3.2rem;
+    font-weight: 1000;
+    padding: 0.35em;
+    text-decoration: none;
+
+    transition: color 0.2s ease-in-out;
+
+    &:hover {
+      color: var(--color-text-primary);
+    }
+
+    &.active {
+      color: var(--color-text-primary);
+    }
+  }
+
+  ${({ $responsive, $open }): string | false =>
+    $responsive &&
+    `
+    flex-flow: column nowrap;
+    align-items: center;
+    justify-content: flex-start;
+    
+    background: #fff;
+    border-radius: 10px;
+    cursor: pointer;
+    height: ${$open ? 'auto' : '95px'}
+    overflow: hidden;
+  `}
+`
+
+const OrderedNavLinkDiv = styled.div<{ $order: number }>`
+  order: ${({ $order }): number => $order};
+`
+
+export interface HeaderNavLinksInterface {
+  label: string
+  withPastLocation?: boolean
+  to: string | Location
+  order: number
+}
+
+const HeaderNavLink: React.FC<HeaderNavLinksInterface & {
+  responsive: boolean
+  showLinkSelector?: boolean
+  handleLinkSelect: (key: NavLinkKeys) => void
+}> = ({
```

Comment:
Can you make an explicit interface for the input params instead of inline declaration?

Reply:
yeah thought i did this :thinking: thanks for catching